### ISSUE TITLE
Too much work downloaded for project limited to 9 tasks at once

### ISSUE THREAD

**Issue by hucker75** [BOINC/boinc#4662]
**Describe the bug**
If I have two projects running, and one is limited to 9 tasks at a time (because it's VB and slows the machine otherwise) and the other project can run on the other 9 cores, the first project downloads an enormous amount of work.  Whenever the second 9 cores are short of work, it asks the first project and gets more work (even though it has plenty) before deciding to use the other project.  It seems to realise those won't go on the second 9 cores, and then asks the 2nd project, but only after it's downloaded them.  My work buffer is 0+1 day, yet I now have enough work to run those 9 cores for 5 days, but they have a 3 day deadline so keep expiring.

**Steps To Reproduce**
1. Attach to Rosetta@Home and Sidock@Home
2. Limit Rosetta@Home to half your cores in its app_config
3. Mine are equal weightings, but to test, set Rosetta at a higher weight than Sidock.
4. Every time Sidock work is finished, it will ask Rosetta first, even though that's already queued up.

**Comment by Toby-Broom** [comment]
It could be the max concurrent bug https://github.com/BOINC/boinc/issues/4322

**Comment by hucker75** [comment]
Agreed.  Somebody can close/delete/combine this thread.  Unfortunately I didn't find the other one first, doesn't seem to be a good way of searching.

**Comment by AenBleidd** [comment]
I'll review this and make some decision. Thank you.

**Comment by AenBleidd** [comment]
Closed as a duplicate of #4322 that is already fixed via #4592
